python/src/uagents/agent.py

```diff
@@ -512,10 +515,18 @@ async def register(self):
         ):
             if self.balance < REGISTRATION_FEE:
                 self._logger.warning(
-                    f"I do not have enough funds to register on Almanac contract\
-                        \nFund using wallet address: {self.wallet.address()}"
+                    "I do not have enough funds to register on Almanac contract"
                 )
-                return
+                if self._test:
+                    add_testnet_funds(str(self.wallet.address()))
+                    self._logger.info(
+                        f"Adding testnet funds to {self.wallet.address()}"
+                    )
+                else:
+                    self._logger.info(
+                        f"Fund using wallet address: {self.wallet.address()}"
```

Comment:
I think "Fund using wallet address:" is confusing as phrase to suggest to fund that wallet. "Add funds by sending FET to current wallet address:"?